src/lib/ngc/compile-source-files.ts

```diff
@@ -53,14 +53,16 @@ export async function compileSourceFiles(
 
   await ngProgram.loadNgStructureAsync();
 
-  const diagnostics = [
+  const allDiagnostics = [
     ...ngProgram.getNgOptionDiagnostics(),
     ...ngProgram.getTsSyntacticDiagnostics(),
     ...ngProgram.getTsSemanticDiagnostics(),
     ...ngProgram.getNgSemanticDiagnostics(),
     ...ngProgram.getNgStructuralDiagnostics()
   ];
 
+  const hasError = ng.exitCodeFromResult(allDiagnostics) > 0;
```

Comment:
Can we move this closer to where it's being read, and change it into:

```suggestion
  const hasError = ng.exitCodeFromResult(allDiagnostics) !== 0;
```